hal/targets/hal/TARGET_NORDIC/TARGET_NRF5/serial_api.c

```diff
@@ -344,64 +344,47 @@ void serial_free(serial_t *obj)
     }
 }
 
-static nrf_uart_baudrate_t baud_translate(int rate)
+void serial_baud(serial_t *obj, int baudrate)
 {
-    nrf_uart_baudrate_t baud;
-
-    if (rate < 57600) {
-        if (rate < 14400) {
-            if (rate < 2400) {
-                baud = NRF_UART_BAUDRATE_1200;
-            } else if (rate < 4800) {
-                baud = NRF_UART_BAUDRATE_2400;
-            } else if (rate < 9600) {
-                baud = NRF_UART_BAUDRATE_4800;
-            } else {
-                baud = NRF_UART_BAUDRATE_9600;
-            }
-        } else {
-            if (rate < 19200) {
-                baud = NRF_UART_BAUDRATE_14400;
-            } else if (rate < 28800) {
-                baud = NRF_UART_BAUDRATE_19200;
-            } else if (rate < 38400) {
-                baud = NRF_UART_BAUDRATE_28800;
-            } else {
-                baud = NRF_UART_BAUDRATE_38400;
-            }
-        }
-    } else {
-        if (rate < 250000) {
-            if (rate < 76800) {
-                baud = NRF_UART_BAUDRATE_57600;
-            } else if (rate < 115200) {
-                baud = NRF_UART_BAUDRATE_76800;
-            } else if (rate < 230400) {
-                baud = NRF_UART_BAUDRATE_115200;
-            } else {
-                baud = NRF_UART_BAUDRATE_230400;
-            }
-        } else {
-            if (rate < 460800) {
-                baud = NRF_UART_BAUDRATE_250000;
-            } else if (rate < 921600) {
-                baud = NRF_UART_BAUDRATE_460800;
-            } else if (rate < 1000000) {
-                baud = NRF_UART_BAUDRATE_921600;
-            } else {
-                baud = NRF_UART_BAUDRATE_1000000;
-            }
+    // nrf_uart_baudrate_set() is not used here (registers are accessed
+    // directly) to make it possible to set special baud rates like 56000
+    // or 31250.
+
+    uint32_t const acceptedSpeeds[][2] = {
```

Comment:
You're right. It should be _static_. It is a _const_ array, so it will (should) be placed in flash. Hence, I thought _static_ would not make any difference. Obviously, it won't hurt.
Apparently, with _static_ added the resulting hex is 16 bytes shorter. I'm not sure why, though.

I added a proper commit.